### ISSUE TITLE
Hangfire retrying faild jobs more than 150 times

### ISSUE THREAD

**Issue by SalmanZafar214** [HangfireIO/Hangfire#1693]
Hi 
i m using hangfire version : 1.6.15 with Asp.net mvc
Storage : Oracle 

![image](https://user-images.githubusercontent.com/6292940/84489170-b2b01480-acba-11ea-9a85-2959857ac048.png)
same worker process has been retried like 300 times . even though stop and restart would not affect
i have used AutomaticRetryAttribute in global.asax as metioned below
![image](https://user-images.githubusercontent.com/6292940/84489303-e8ed9400-acba-11ea-91b2-52054b333a0d.png)

issue that i m facing is 
![image](https://user-images.githubusercontent.com/6292940/84489378-11758e00-acbb-11ea-8b93-fdf8b9a3b513.png)

This issue is logged multiple times count is more than 1000 . 
I need to sort out this issue as its hapening on our production server 
connection string for oracle is 

`<add name="AdocsEntities" connectionString="Data Source=(DESCRIPTION = (ADDRESS = (PROTOCOL = TCP)(HOST = [DATABASE_IP])(PORT = [PORT_NUMBER]))(CONNECT_DATA =(SERVER = DEDICATED)(SERVICE_NAME = orcl)));User ID=[USER_ID];Password=[PASSWORD];Pooling=true;Min Pool Size=0;Max Pool Size=100;Connection Lifetime=0" providerName="oracle.manageddataaccess.client"/> `

i have google and come to know that Min Pool size needs to be increased with Timeout even though i dont know it will have some affect or not . 
Can any one help me how can i fix the above issue if its not possible can i stop retries as in my case RetryAttribute not working :(

**Comment by DaveSweeton** [comment]
I'm seeing extensive retries on a method that has a retry attribute: [AutomaticRetry(Attempts = 2)] The dashboard shows "Retry attempt 2 of 2", but it's trying the job over and over.

I'm using SQL server, and 1.7.13

**Comment by odinserj** [comment]
Dave please enable logging to always know about all the exceptions – https://docs.hangfire.io/en/latest/configuration/configuring-logging.html – and answer #1741, looks like your schema was altered and `PK_HangFire_JobParameter` index is not unique.

**Comment by DaveSweeton** [comment]
Is this also fixed by bf93046 and the upcoming 1.7.14? Sounds like they were related.

**Comment by odinserj** [comment]
Yes Dave, this problem was also fixed for your case (but does not relate to Oracle). 1.7.14 already released, so you can grab an update.

**Comment by DaveSweeton** [comment]
Yep, 1.7.14 fixes my retry issue (on Sql Server). Thanks for the quick fix!
